src/NuGet.Core/NuGet.Protocol/Resources/PackageSearchResourceV3.cs

```diff
@@ -1,22 +1,40 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
 using System.Linq;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
 using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Model;
 
 namespace NuGet.Protocol
 {
     public class PackageSearchResourceV3 : PackageSearchResource
     {
-        private readonly RawSearchResourceV3 _rawSearchResource;
+        private readonly HttpSource _client;
+        private readonly Uri[] _searchEndpoints;
 
-        public PackageSearchResourceV3(RawSearchResourceV3 searchResource)
```

Comment:
We need to keep the old constructor and make a switch in the code based on which constructor was used. 

That was consumers like @campersau are not broken in any way.

Reply:
Ok. I added back old constructor along new one.